core/src/main/java/org/fao/geonet/api/records/attachments/CMISStore.java

```diff
@@ -506,7 +506,7 @@ public MetadataResourceContainer getResourceContainerDescription(final ServiceCo
             if (folderRoot == null) {
                 folderRoot = "";
             }
-            Folder parentFolder = cmisUtils.getFolderCache(key + folderRoot);
+            Folder parentFolder = cmisUtils.getFolderCache(key + folderRoot, false, true);
```

Comment:
@wangf1122 

I don't think we want it to create folders when it is not found as it could end up creating a lot of unused folders.

I think the issue is the following line 516

`} catch (CmisObjectNotFoundException e) {`

I think it should be 

`} catch (CmisObjectNotFoundException | ResourceNotFoundException e) {`

because `getFolderCache `throws `ResourceNotFoundException` and I think it used to throw `CmisObjectNotFoundException` but in either case they are similar exceptions.

Can you test the suggested change.

Reply:
Looking at the filesystemStore, it seems to always create the folder.

https://github.com/geonetwork/core-geonetwork/blob/c3fe382efb3d7fccd576da319abea09cab535a7f/core/src/main/java/org/fao/geonet/api/records/attachments/FilesystemStore.java#L181-L195

So maybe you are correct and maybe we should remove the exception handling for `CmisObjectNotFoundException`

Reply:
I haven't commit the code for exception handle. So this code will be fine here. Yeah, i also tested the file system and they do create resource hiddenly in the same implementation method.